### PR TITLE
ENH: integrate: support ILP64 BLAS in odeint/vode/lsoda

### PR DESCRIPTION
**Pull request by pv**
#### Reference issue
Part of gh-11193

#### What does this implement/fix?
Support ILP64 BLAS in odeint/vode/lsoda.
Also convert dop to 64-bit Fortran ints, even though it doesn't use blas.

### CODE REVIEW

File: scipy/integrate/setup.py [hunk 1]
```diff
@@ -28,15 +34,25 @@ def configuration(parent_package='',top_path=None):
     quadpack_test_src = [join('tests','_test_multivariate.c')]
     odeint_banded_test_src = [join('tests', 'banded5x5.f')]
 
+    if uses_blas64():
+        mach64_name = 'mach64'
+        config.add_library('mach64', sources=mach_src,
+                           config_fc={'noopt':(__file__,1)})
```
Comment (rgommers) [inline scipy/integrate/setup.py:40]: The new `lint_diff.py` tool wants a space after `:` here.

File: scipy/integrate/setup.py [hunk 2]
```diff
@@ -28,15 +34,25 @@ def configuration(parent_package='',top_path=None):
...
-    config.add_library('lsoda', sources=lsoda_src)
-    config.add_library('vode', sources=vode_src)
-    config.add_library('dop', sources=dop_src)
+    config.add_library('lsoda', sources=lsoda_src, _pre_build_hook=pre_build_hook)
+    config.add_library('vode', sources=vode_src, _pre_build_hook=pre_build_hook)
+    config.add_library('dop', sources=dop_src, _pre_build_hook=pre_build_hook)
 
     # Extensions
     # quadpack:
+    lapack_opt = dict(get_info('lapack_opt',notfound_action=2))
```
Comment (rgommers) [inline scipy/integrate/setup.py:54]: and a space after comma here

Comment (pv) [reply scipy/integrate/setup.py]: fixed

**Comment by rgommers** [comment]
LGTM, merged. Thanks @pv

**Pull request merged**